hadoop-client-modules/hadoop-client-integration-tests/pom.xml

```diff
@@ -52,6 +52,16 @@
       <artifactId>junit</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```

Comment:
This doesn't look like the right approach, since the integration tests shouldn't depend on test dependency. As mentioned earlier, I think the issue is that the `snappy-java` is included in the optional dependency `hadoop-client` so it won't be a transitive dependency for downstream apps. Therefore, those apps will have to explicitly declare `snappy-java` as a dependency, which could cause surprises.

Instead I think we should add `snappy-java` as compile dependency in `hadoop-client-api`. You may need to fight the maven enforcer rules for this.

Reply:
Oh I see. It makes sense.

Reply:
I just tested it locally. Seems working. Let me change it and see CI result.